### DISCUSSION TITLE
classes/class_string

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#165] [User-contributed notes]
# classes/class_string

A built-in type for strings. Description: This is the built-in string Variant type (and the one used by GDScript). Strings may contain any number of Unicode characters, and expose methods useful fo...

https://docs.godotengine.org/en/stable/classes/class_string.html

**Comment by maxd66** [comment]
It is not entirely clear, but the replace method (and I'm assuming several others that I haven't tested) return a new string, they do not alter the original.

I know that is often the case and probably assumed by most, but just wanted to make that clear for anyone that might not have found that obvious.

**Reply by Calinou** [reply]
String is immutable in Godot, so all methods return a new String and never modify the original String.

**Comment by The5-1** [comment]
the `to_int()` function interestingly can handle thousands separators correctly:

`var e = "1_000_000.05".to_int() # e is 1000000`

And since I think I relates, there also this proposal for offering thousands formatting: https://github.com/godotengine/godot-proposals/issues/4971
